nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -392,18 +388,28 @@ def training_step(self, batch, batch_idx):
         if self.cfg.precision == 16:
             loss_scale = self.trainer.precision_plugin.scaler._scale
             if loss_scale is not None:
-                self.log('loss_scale', loss_scale)
+                self.log('loss_scale', loss_scale, batch_size=self.cfg.micro_batch_size)
 
-        self.log('reduced_train_loss', loss_mean, prog_bar=True, rank_zero_only=True)
+        self.log(
+            'reduced_train_loss', loss_mean, prog_bar=True, rank_zero_only=True, batch_size=self.cfg.micro_batch_size
+        )
         lr = self._optimizer.param_groups[0]['lr']
-        self.log('lr', lr, rank_zero_only=True)
-        self.log('global_step', self.trainer.global_step, prog_bar=True, rank_zero_only=True)
+        self.log('lr', lr, rank_zero_only=True, batch_size=self.cfg.micro_batch_size)
+        self.log(
+            'global_step',
+            self.trainer.global_step,
+            prog_bar=True,
+            rank_zero_only=True,
+            batch_size=self.cfg.micro_batch_size,
+        )
```

Comment:
Why do we add batch_size argument to self.log? And should it be micro_batch_size or global_batch_size?